### PR TITLE
[FLINK-16901][legal] Correctly handle the THIRD_PARTY_NOTICES file in kinesis connector bundled dependency

### PR DESCRIPTION
**Pull request by carp84**
## What is the purpose of the change

Fix the legal issue in kinesis connector as reported by FLINK-16901.

## Brief change log

Exclude `THIRD_PARTY_NOTICES` when shading `amazon-kinesis-producer` and append the content of this file into `META-INF/NOTICES` of the `flink-connector-kinesis` module, following instructions of https://cwiki.apache.org/confluence/display/FLINK/Licensing

This also adds the missing `com.amazonaws:aws-java-sdk-dynamodb` dependency in `META-INF/NOTICE`

## Verifying this change

Manually check the generated `flink-connector-kinesis` jar file and confirm there's no `THIRD_PARTY_NOTICES` included but its content appears in `META-INF/NOTICE`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by carp84** [comment]
@zentol @rmetzger could you help review this one? Thanks.

File: flink-connectors/flink-connector-kinesis/pom.xml [hunk 1]
```diff
@@ -221,6 +221,14 @@ under the License.
 									<shadedPattern>org.apache.flink.kinesis.shaded.org.apache.http</shadedPattern>
 								</relocation>
 							</relocations>
+							<filters>
+								<filter>
+									<artifact>com.amazonaws:amazon-kinesis-producer</artifact>
+									<excludes>
+										<exclude>META-INF/THIRD_PARTY_NOTICES</exclude>
```
Comment (zentol) [inline flink-connectors/flink-connector-kinesis/pom.xml:228]: it's so annoying that there isn't a standard location for this stuff...

Comment (carp84) [reply flink-connectors/flink-connector-kinesis/pom.xml:228]: Agreed...

**Comment by carp84** [comment]
Thanks for review @zentol , will merge soon.

**Pull request closed without merging**